plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlValidator.java

```diff
@@ -156,30 +155,30 @@ private void validateServletMapping() {
   private void validateJsp() {
     IProject project = resource.getProject();
     IVirtualComponent component = ComponentCore.createComponent(project);
+    // TODO: Search for JSP files in jars if web.xml is version 3.0+.
+    // JSP file META-INF/resources/test.jsp in a jar should be able to be referenced
+    // as test.jsp in web.xml.
     if (component != null && component.exists()) {
+      // For a typical Maven project:
+      // WEB-INF         -> src/main/webapp/WEB-INF
+      //    /            -> src/main/webapp
+      // WEB-INF/classes -> src/main/java
+      // WEB-INF/lib     -> src/main/webapp/WEB-INF/lib
       IVirtualFolder root = component.getRootFolder();
       if (root.exists()) {
         NodeList jspList = document.getElementsByTagName("jsp-file");
         for (int i = 0; i < jspList.getLength(); i++) {
           Node jspNode = jspList.item(i);
           String jspName = jspNode.getTextContent();
           IFile file = root.getFile(jspName).getUnderlyingFile();
```

Comment:
Would separating our the resolve-JSP into a separate method (`IFile resolveJsp(IVirtualComponent, String jspPath)`) simplify this method?